include/logging/log_backend.h

```diff
@@ -48,6 +48,8 @@ struct log_backend_api {
 	void (*dropped)(const struct log_backend *const backend, uint32_t cnt);
 	void (*panic)(const struct log_backend *const backend);
 	void (*init)(const struct log_backend *const backend);
+	int (*format_set)(const struct log_backend *const backend,
```

Comment:
why size_t? This just an integer indicating the type.

Reply:
@nashif log_format_table_size() was changed to return type "size_t" as I think it is a general good practice returning a size should use size_t (when the ARRAY_SIZE operator is involved) . 

Since there were comparisons in the API log_backend_format_set [ log_type >= log_format_table_size() ] and also we do not want a negative value for log_type therefore considered size_t type for log_type. Let me know if this doesn't make sense?

Reply:
> @nashif log_format_table_size() was changed to return type "size_t" as I think it is a general good practice returning a size should use size_t (when the ARRAY_SIZE operator is involved) .

Using it for log_format_table_size is fine, I am wondering if using it here...
```
static size_t log_format_current = CONFIG_LOG_BACKEND_ADSP_OUTPUT_DEFAULT;
```

and elsewhere is the right choice of type, although it makes sense to use size_t for array indexes, but this is not clear in the above code for example.

Reply:
I have changed the size_t type to uint32_t in all the places except for log_format_table_size where I still use "size_t".